src/ui/Modal/FormikModal.tsx

```diff
@@ -0,0 +1,201 @@
+import { IModalProps, Modal, useDialogSize } from ".";
+import { Form, Formik } from "formik";
+import { Button } from "../Buttons/Button";
+import { Fragment, createContext, useContext, useEffect, useRef, useState } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { XIcon } from "@heroicons/react/solid";
+import clsx from "clsx";
+import { BsArrowsFullscreen, BsFullscreenExit } from "react-icons/bs";
+import DialogButton from "../Buttons/DialogButton";
+import HelpLink from "../Buttons/HelpLink";
+
+interface FormikModalProps extends IModalProps {
+  initialValues?: Record<string, any>;
```

Comment:
This in context will be ineffective, as initial value an only be set on initial render, and the children will have access to all formik controls, to update the values, so they can update the values directly instead, if they need to.